Code/Source/solver/fsi.cpp

```diff
@@ -320,6 +327,12 @@ void construct_fsi(ComMod& com_mod, CepMod& cep_mod, const mshType& lM, const Ar
       }
     } // g: loop
 
+
+    if (cPhys == Equation_ustruct) 
+    {
+      ustruct::ustruct_do_assem(com_mod, eNoN, ptr, lKd, lK, lR);
+    }
```

Comment:
Are you assembling twice, calling `ustruct::ustruct_do_assem()` and then `eq.linear_algebra->assemble()` ?